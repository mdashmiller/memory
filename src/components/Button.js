import styled from 'styled-components'

const Button = styled.button`
	height: 40px;
	width: 80px;
	text-align: center;
	border-radius: 7px;
	margin: 0 auto;
	display: block;
	border: none;
	font-size: 1em;
	font-family: Arial;
	background: yellow;
	color: red;
	font-weight: 700;
`

export default Button
