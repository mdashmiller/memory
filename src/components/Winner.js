import React from 'react'
import styled from 'styled-components'
import Button from './Button'

const Banner = styled.div`
	opacity: 0.9;
    background: black;
    height: 290px;
    width: 320px;
    text-align: center;
    color: #fccdd3;
    margin: 14px auto;
    position: absolute;
    border-radius: 4px
`

const BannerTitle = styled.p`
	font-size: 44px;
    font-weight: 700;
    margin-top: 20px;
    margin-bottom: 63px;
`

const Score = styled.p`
	font-size: 26px;
    margin: 0 auto;
    color: #62bcfa;
`

const BannerSubTitle = styled.p`
	font-size: 30px;
	margin: 0 auto;
`

const Winner = ({
	moves,
	newBest,
	replay
}) =>
	<Banner>
		<BannerTitle>Winner!!!</BannerTitle>
		<Score>Moves: {moves}</Score>
		{newBest &&
			<BannerSubTitle className="flicker">
				New Record!
			</BannerSubTitle>
		}
		<Button
			win
			onClick={replay}
		>
			Play Again?
		</Button>
	</Banner>

export default Winner
