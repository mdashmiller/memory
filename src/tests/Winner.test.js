import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import Winner from '../components/Winner'
import Button from '../components/Button'

Enzyme.configure({ adapter: new Adapter() })

const props = {
	moves: 5,
	newBest: true,
	replay: jest.fn()
}

describe('<Winner />', () => {

	let wrapper

	beforeAll(() => {
		wrapper = shallow(<Winner { ...props } />)
	})

	it('renders without crashing', () => {
		const div = document.createElement('div')
		ReactDOM.render(<Winner { ...props } />, div)
		ReactDOM.unmountComponentAtNode(div)
	})

	test('has a valid snapshot', () => {
		const component = renderer.create(
			<Winner { ...props } />
		)
		let tree = component.toJSON()
		expect(tree).toMatchSnapshot()
	})

	it('renders 1 div #banner', () => {
		expect(wrapper.find('#banner').length).toBe(1)
	})

	it('renders 1 p #banner-title', () => {
		expect(wrapper.find('#banner-title').length).toBe(1)
	})

	it('renders 1 p #score', () => {
		expect(wrapper.find('#score').length).toBe(1)
	})

	it('renders 1 <Button /> component', () => {
		expect(wrapper.find(Button).length).toBe(1)
	})

	it('displays moves in #score', () => {
		expect(wrapper.find('#score').props().children[1]).toBe(5)
	})

	it('renders 1 bannerSubTitle when new best score exists', () => {
		expect(wrapper.find('.flicker').length).toBe(1)
	})

	it('handles click events', () => {
		wrapper.find(Button).simulate('click')

		expect(props.replay).toHaveBeenCalled()
	})

})
