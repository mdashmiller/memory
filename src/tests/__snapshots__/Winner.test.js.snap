// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Winner /> has a valid snapshot 1`] = `
<div
  className="sc-bwzfXH gaSrbo"
  id="banner"
>
  <p
    className="sc-htpNat lhATxg"
    id="banner-title"
  >
    Winner!!!
  </p>
  <p
    className="sc-bxivhb jjmwiE"
    id="score"
  >
    Moves: 
    5
  </p>
  <p
    className="flicker sc-ifAKCX gvkNLs"
  >
    New Record!
  </p>
  <button
    className="sc-bdVaJa eknjhX"
    onClick={[MockFunction]}
  >
    Play Again?
  </button>
</div>
`;
